### PR TITLE
Span validator to log incase SDK sends wrong timespan

### PR DESCRIPTION
**Pull request by agarwalmukund**
## Proposed changes:
- Added a span validator which is called during encode process.
- Added test to check the validator.

### Pre-requisites:
- [x] Reviewed code before asking reviewer <em>(Look at *Files Changed* below instead of Xcode)</em>.
- [x] Reviewed code format follows the style guide where possible.
- [x] Added tests where possible <em>(*compulsory for business logic and new classes)</em>.

### CODE REVIEW

File: Trace/TraceModel/TraceModel+Span.swift [hunk 1]
```diff
@@ -301,6 +306,32 @@ extension TraceModel {
             
             return copy
         }
+        
+        // MARK: - Span validator
+        
+        /// :nodoc:
+        @discardableResult
+        public func spanValidator(span: Span) -> Bool {
```
Comment (shams-ahmed) [inline Trace/TraceModel/TraceModel+Span.swift]: 1. This needs to be an internal method as it is for internal use only
2. I don't understand why it requires a span object when it's an instance method. wouldn't `func validate() -> Bool` do the same. 
3. if `.end` is not present shouldn't that return false as it isn't in the correct state.

File: Tests/TraceModelTests.swift [hunk 2]
```diff
@@ -337,4 +337,55 @@ final class TraceModelTests: XCTestCase {
         XCTAssertEqual(model.spans.count, 5002)
         XCTAssertTrue(model.isComplete)
     }
+    
+    func testTraceSpan_valid() {
```
Comment (shams-ahmed) [inline Tests/TraceModelTests.swift:341]: NIT: 
1. Add another test where the start and end time are the same.  
2. Add a test for when the end time is not present

**Review by shams-ahmed — CHANGES_REQUESTED** [review]

File: Trace/TraceModel/TraceModel+Span.swift (hunk 1 above)
Comment (agarwalmukund) [reply Trace/TraceModel/TraceModel+Span.swift]: Thanks all valid and makes sense to me. Updated.

File: Tests/TraceModelTests.swift (hunk 2 above)
Comment (agarwalmukund) [reply Tests/TraceModelTests.swift:341]: Thanks. Updated.

File: Trace/TraceModel/TraceModel+Span.swift [hunk 3]
```diff
@@ -301,6 +306,31 @@ extension TraceModel {
...
+        internal func validate() -> Bool {
+            guard let strongEnd = end else {
+                return false
+            }
+            
+            var valid = true
+            if start.seconds > strongEnd.seconds {
+                valid = false
+            } else if start.seconds == strongEnd.seconds {
+                if start.nanos > strongEnd.nanos || start.nanos == strongEnd.nanos {
```
Comment (shams-ahmed) [inline Trace/TraceModel/TraceModel+Span.swift]: Can you use `>=` as it's easier to read? This way you would only need one check

**Review by shams-ahmed — APPROVED** [review]
Minor suggestions

File: Trace/TraceModel/TraceModel+Span.swift (hunk 3 above)
Comment (agarwalmukund) [reply Trace/TraceModel/TraceModel+Span.swift]: Thanks updated.

**Pull request merged**